tests/functional/test_install.py

```diff
@@ -225,72 +225,38 @@ def test_install_dev_version_from_pypi(script):
     )
 
 
-@pytest.mark.network
 def test_install_editable_from_git(script, tmpdir):
-    """
-    Test cloning from Git.
-    """
-    args = ['install']
-    args.extend([
-        '-e',
-        '%s#egg=pip-test-package' %
-        local_checkout(
-            'git+http://github.com/pypa/pip-test-package.git',
-            tmpdir.join("cache"),
-        ),
-    ])
+    """Test cloning from Git."""
+    pkg_path = _create_test_package(script, name='testpackage', vcs='git')
+    args = ['install', '-e', 'git+%s#egg=testpackage' % path_to_url(pkg_path)]
     result = script.pip(*args, **{"expect_error": True})
-    result.assert_installed('pip-test-package', with_files=['.git'])
+    result.assert_installed('testpackage', with_files=['.git'])
 
 
-@pytest.mark.network
 def test_install_editable_from_hg(script, tmpdir):
-    """
-    Test cloning from Mercurial.
-    """
-    result = script.pip(
-        'install', '-e',
-        '%s#egg=ScriptTest' %
-        local_checkout(
-            'hg+https://bitbucket.org/ianb/scripttest',
-            tmpdir.join("cache"),
-        ),
-        expect_error=True,
-    )
-    result.assert_installed('ScriptTest', with_files=['.hg'])
+    """Test cloning from Mercurial."""
+    pkg_path = _create_test_package(script, name='testpackage', vcs='hg')
+    args = ['install', '-e', 'hg+%s#egg=testpackage' % path_to_url(pkg_path)]
+    result = script.pip(*args, **{"expect_error": True})
+    result.assert_installed('testpackage', with_files=['.hg'])
 
 
-@pytest.mark.network
 def test_vcs_url_final_slash_normalization(script, tmpdir):
     """
     Test that presence or absence of final slash in VCS URL is normalized.
     """
-    script.pip(
-        'install', '-e',
-        '%s/#egg=ScriptTest' %
-        local_checkout(
-            'hg+https://bitbucket.org/ianb/scripttest',
-            tmpdir.join("cache"),
-        ),
-    )
```

Comment:
It took me a while to see it too. I think what it's testing is the `/` after `%s` and before `#egg=`. I guess there is normalization code that strips that out. That's my guess at least.